lib/model/message_list.dart

```diff
@@ -487,6 +507,92 @@ class MessageListView with ChangeNotifier, _MessageSequence {
     }
   }
 
+  void _messagesMovedInternally(List<int> messageIds) {
+    for (final messageId in messageIds) {
+      if (_findMessageWithId(messageId) != -1) {
+        _reprocessAll();
+        notifyListeners();
+        return;
+      }
+    }
+  }
+
+  void _messagesMovedIntoNarrow() {
+    // If there are some messages we don't have in [MessageStore], and they
+    // occur later than the messages we have here, then we just have to
+    // re-fetch from scratch.  That's always valid, so just do that always.
+    // TODO in cases where we do have data to do better, do better.
+    _reset();
+    notifyListeners();
+    fetchInitial();
+  }
+
+  void _handlePropagateMode(PropagateMode propagateMode, Narrow newNarrow) {
+    switch (propagateMode) {
+      case PropagateMode.changeAll:
+      case PropagateMode.changeLater:
+        narrow = newNarrow;
+        _reset();
+        fetchInitial();
+      case PropagateMode.changeOne:
+    }
+  }
+
+  void _messagesMovedFromNarrow(
+    List<int> messageIds,
+    PropagateMode propagateMode,
+    {TopicNarrow? newNarrow}
+  ) {
+    if (_removeMessagesById(messageIds)) {
+      if (newNarrow != null) {
+        _handlePropagateMode(propagateMode, newNarrow);
```

Comment:
> Fixing that would be complex and is out of scope at present, even if we were to decide it's desirable.

Starting a [conversation](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Accurately.20handle.20propagateMode.2EchangeLater) so that we have a place for future discussion on this.